fluss-common/src/main/java/com/alibaba/fluss/row/encode/paimon/PaimonBinaryRowWriter.java

```diff
@@ -0,0 +1,420 @@
+/*
+ * Copyright (c) 2025 Alibaba Group Holding Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.fluss.row.encode.paimon;
+
+import com.alibaba.fluss.memory.MemorySegment;
+import com.alibaba.fluss.record.RowKind;
+import com.alibaba.fluss.row.BinarySegmentUtils;
+import com.alibaba.fluss.row.BinaryString;
+import com.alibaba.fluss.row.Decimal;
+import com.alibaba.fluss.row.InternalRow;
+import com.alibaba.fluss.row.TimestampLtz;
+import com.alibaba.fluss.row.TimestampNtz;
+import com.alibaba.fluss.types.DataType;
+import com.alibaba.fluss.utils.UnsafeUtils;
+
+import java.io.Serializable;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+
+import static com.alibaba.fluss.types.DataTypeChecks.getPrecision;
+
+/**
+ * A writer to encode Fluss's {@link InternalRow} using Paimon's BinaryRow encoding way.
+ *
+ * <p>The logic is almost copied from Paimon's BinaryRowWriter.
+ */
+class PaimonBinaryRowWriter {
+
+    private static final int HEADER_SIZE_IN_BITS = 8;
+    private static final int MAX_FIX_PART_DATA_SIZE = 7;
+
+    private final int nullBitsSizeInBytes;
+    private final int fixedSize;
+    private byte[] buffer;
+    private MemorySegment segment;
+
+    private int cursor;
+
+    public PaimonBinaryRowWriter(int arity) {
+        this.nullBitsSizeInBytes = calculateBitSetWidthInBytes(arity);
+        this.fixedSize = getFixedLengthPartSize(nullBitsSizeInBytes, arity);
+        this.cursor = fixedSize;
+        setBuffer(new byte[fixedSize]);
+    }
+
+    public void reset() {
+        this.cursor = fixedSize;
+        for (int i = 0; i < nullBitsSizeInBytes; i += 8) {
+            UnsafeUtils.putLong(buffer, i, 0L);
+        }
+    }
+
+    public byte[] toBytes() {
+        byte[] bytes = new byte[cursor];
+        System.arraycopy(buffer, 0, bytes, 0, cursor);
+        return bytes;
+    }
+
+    public void setNullAt(int pos) {
+        setNullBit(pos);
+        UnsafeUtils.putLong(buffer, getFieldOffset(pos), 0L);
+    }
+
+    private void setNullBit(int pos) {
+        UnsafeUtils.bitSet(buffer, 0, pos + HEADER_SIZE_IN_BITS);
+    }
+
+    public void writeRowKind(RowKind kind) {
+        // Fluss has APPEND_ONLY rowKind, so minus 1 to align with paimon
```

Comment:
Convert Fluss RowKind into Paimon RowKind. If the kind is `APPEND_ONLY`, there may be unpredictable problem to use `-1` kind byte here.